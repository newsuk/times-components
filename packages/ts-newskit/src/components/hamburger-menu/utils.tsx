import React from 'react';
import { Story } from '@storybook/addon-docs';
import { ThemeProvider } from 'newskit';
import { TimesWebLightTheme } from '../../theme';

export const StoryWithTheme: React.FC<{ name: string }> = ({
  name,
  children
}) => (
  <ThemeProvider theme={TimesWebLightTheme}>
    <Story name={name}>{children}</Story>
  </ThemeProvider>
);
