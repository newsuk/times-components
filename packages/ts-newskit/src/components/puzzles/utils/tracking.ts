import { PuzzleScrollClickHandlerType, MouseEventType } from '../cards-container/types';

type TrackingPuzzleCategoryProps = {
  nextPrev: string;
  puzzleType: string;
};

export const puzzleCategoryClickTracking = (
  event: MouseEventType,
  puzzleScroll: TrackingPuzzleCategoryProps,
  clickHandler: PuzzleScrollClickHandlerType
) => {
  puzzleScroll && clickHandler(event, puzzleScroll);
};
