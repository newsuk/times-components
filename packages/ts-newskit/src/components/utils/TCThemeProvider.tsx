import React from 'react';
import { NewsKitProvider, Theme } from 'newskit';
import { TimesWebLightTheme } from '../../theme';

type TCThemeProviderProps = {
  theme?: Theme;
};

export const TCThemeProvider: React.FC<TCThemeProviderProps> = ({
  theme = TimesWebLightTheme,
  children
}) => <NewsKitProvider theme={theme}>{children}</NewsKitProvider>;
