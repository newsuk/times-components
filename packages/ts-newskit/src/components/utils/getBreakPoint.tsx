import { useBreakpointKey } from 'newskit';
import { SecondaryMenuItem } from '../secondary-menu/types';

export const getBreakpoint = (data: SecondaryMenuItem[]) => {
  const breakpointKey = useBreakpointKey();

  const menuItems =
    breakpointKey === 'md' ? 7 : breakpointKey === 'lg' ? 9 : data.length;

  const moreMenuLength = data.length - menuItems;

  return { moreMenuLength, menuItems, breakpointKey };
};
