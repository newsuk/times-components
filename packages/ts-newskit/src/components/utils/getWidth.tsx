export const getWidth = (el: any) => el.clientWidth;
