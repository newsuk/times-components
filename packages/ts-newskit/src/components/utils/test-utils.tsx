import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from 'newskit';
import { TimesWebLightTheme } from '../../theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={TimesWebLightTheme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
