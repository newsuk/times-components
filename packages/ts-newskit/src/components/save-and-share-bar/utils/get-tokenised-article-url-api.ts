// @ts-ignore
import { getTokenisedArticleUrl } from '@times-components/provider-queries';
// @ts-ignore
import { makeClient } from '@times-components/utils';

const client = makeClient();

export const getTokenisedArticleUrlApi = (id: string) =>
  client.query({ query: getTokenisedArticleUrl, variables: { id } });
