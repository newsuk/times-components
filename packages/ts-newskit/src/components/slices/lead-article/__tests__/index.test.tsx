import React from 'react';
import '@testing-library/jest-dom';
import { render } from '../../../../utils/test-utils';
import { LeadArticle, LeadArticleProps } from '../index';
import { leadArticle } from '../../../../slices/fixtures/data.json';
import { useBreakpointKey } from 'newskit';

jest.mock('newskit', () => ({
  ...jest.requireActual('newskit'),
  useBreakpointKey: jest.fn().mockReturnValue('xs')
}));

const leadStoryData: LeadArticleProps = {
  ...leadArticle
};

const renderComponent = () =>
  render(<LeadArticle {...leadStoryData} imageTop={false} />);

describe('Render Component one', () => {
  it('should render a snapshot', () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render correct headline', () => {
    const { getByText } = renderComponent();
    const headlineText = getByText(leadArticle.headline);
    expect(headlineText).toBeInTheDocument();
  });

  it('should render correct summary', () => {
    const { getByText } = renderComponent();

    const summaryText = getByText(leadArticle.summary);
    expect(summaryText).toBeInTheDocument();
  });
  it('should render correct readingTime', () => {
    const { getByText } = renderComponent();

    const readingTimeText = getByText(leadArticle.tag.label);
    expect(readingTimeText).toBeInTheDocument();
  });
  it('should render correct caption', () => {
    const { getByText } = renderComponent();

    const captionText = getByText(leadArticle.images.caption);
    expect(captionText).toBeInTheDocument();
  });

  it('should render tag if tag is provided', () => {
    (useBreakpointKey as any).mockReturnValue('xs');
    const { queryByText } = render(
      <LeadArticle {...leadStoryData} imageTop={true} />
    );
    const tag = queryByText(leadArticle.tag.label);
    expect(tag).toBeVisible();
  });

  it('should not render tag or flag if they are not provided', () => {
    (useBreakpointKey as any).mockReturnValue('lg');

    delete leadStoryData.tag;
    delete leadStoryData.flag;

    const { queryByText } = render(<LeadArticle {...leadStoryData} />);
    const tag = queryByText('Tag');
    const flag = queryByText('Flag');

    expect(tag).not.toBeInTheDocument();
    expect(flag).not.toBeInTheDocument();
  });

  it('should use corrrect spacing if `contentTop` is set', () => {
    const { getByText } = render(<LeadArticle {...leadStoryData} contentTop />);
    const headline = getByText(leadStoryData.headline);
    const cardContainer = headline.closest('div');

    expect(cardContainer).toHaveStyle({
      marginBlockEnd: '16px'
    });
  });

  it('should render ordered list without link  if href not provided', () => {
    const modifiedLeadStoryData: LeadArticleProps = {
      ...leadArticle,
      listData: [{ label: 'Unordered list item', href: '' }]
    };
    const { getByText } = render(
      <LeadArticle {...modifiedLeadStoryData} contentTop />
    );
    const link = getByText('Unordered list item');
    expect(link.tagName.toLowerCase()).toBe('p');
  });
});
