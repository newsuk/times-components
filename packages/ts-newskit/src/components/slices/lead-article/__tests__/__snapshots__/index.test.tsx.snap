// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Component one should render a snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-19qmu5v"
  >
    <div
      class="css-1vuk75f"
    >
      <div
        class="css-uxw9nr"
      >
        <picture
          class="css-zp0elg"
        >
          <div
            class="css-5pczz7"
          />
          <img
            alt="This is ALT Text"
            class="css-1qgqq71"
            src="/assets/main_01.jpg"
          />
        </picture>
      </div>
      <p
        class="css-9pk24b"
      >
        Credit
      </p>
    </div>
    <div
      class="css-gpke1l"
    >
      <hr
        aria-hidden="true"
        class="css-tk98gp"
        data-testid="divider"
      />
      <a
        class="css-10h0i3c"
        href="https://www.thetimes.co.uk"
      >
        <span
          class="css-17x5lw"
        >
          <span
            class="css-8gj81n"
          >
            Sarcacens an inclusive club? They didnt look out for me
          </span>
        </span>
      </a>
      <p
        class="css-1nrveqv"
      >
        Lyon lock Joel Kpoku tells Will Kelleher that he’s fired up to face old team — and urges more English stars to move to France
      </p>
      <div
        class="css-1atazsr"
        data-testid="tag-and-flag"
      >
        <a
          class="css-10d4t8c"
          href="/"
        >
          <span
            class="css-17x5lw"
          >
            <span
              class="css-zhtub2"
            >
              Review
            </span>
          </span>
        </a>
        <div
          class="css-lew009"
        >
          <hr
            aria-hidden="true"
            class="css-15kf7hd"
            data-testid="divider"
          />
        </div>
        <span
          class="css-1vzvkql"
        >
          4 min read
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
